### PR TITLE
Doctest

### LINKED ISSUE #121: Add testing to documentation code
#120 demonstrates the need to have tests for our examples in the documentation.

### PR DESCRIPTION
**Pull request by bknueven**
## Fixes/Addresses:

Fixes #121 

## Summary/Motivation:

We should be testing the code in our documentation

## Changes proposed in this PR:
- Adds doctest to GitHub actions
- Set up testing for existing documentation

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/how_to/how_to_setup_RO_config_options.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ contains options that account for concentration polarization, the mass transfer
 Example: Configure the RO model to account for concentration polarization and pressure drop
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. code-block:: python
+.. testcode::
```
Comment (adam-a-a) [inline docs/how_to/how_to_setup_RO_config_options.rst:9]: @bknueven When should we use the `testcode` directive vs. the `testsetup` directive? I noticed that in `how_to_scale_a_model.rst`, `testsetup` was used for a portion of code similar (with a few more lines of code) to this code block.

Comment (bknueven) [reply docs/how_to/how_to_setup_RO_config_options.rst:9]: The `testsetup` directive runs, but doesn't display the code, for just the situation where you don't want the user to see the setup to focus on the task being described.

The `testcode` directive displays the code in the documentation, so I replaced `code-block:: python` everywhere I found it with `testcode` so the look of documentation didn't change with this PR. I don't mind the additional imports in the RO config, but they could be done away with as you suggest.

Comment (adam-a-a) [reply docs/how_to/how_to_setup_RO_config_options.rst:9]: Thanks for the explanation! 

I think in this particular case, it is useful to show the extra imports needed for the RO config to show the user how to configure RO to account for concentration polarization and pressure drop. So, I don't think you should do away with them here.

**Review by adam-a-a — APPROVED** [review]
Looks good to me! I wasn't sure how to go about testing documentation, so thanks for the lesson!

**Comment by bknueven** [comment]
I'd like to bring in the parameter sweep how-to (#123) before this is merged

File: docs/technical_reference/property_models/NaCl.rst [hunk 2]
```diff
@@ -78,11 +78,16 @@ This NaCl property package includes support for scaling, such as providing defau
...
+
+   from pyomo.environ import ConcreteModel
+   from idaes.core import FlowsheetBlock
+
+.. testcode::
    
    # relevant imports
    import proteuslib.property_models.NaCl_prop_pack as props
-   import idaes.core.util.scaling as calculate_scaling_factors
+   from idaes.core.util.scaling import calculate_scaling_factors
```
Comment (TimBartholomew) [inline docs/technical_reference/property_models/NaCl.rst:90]: And this is why we need testing :)
Good catch.

**Review by TimBartholomew — DISMISSED** [review]
@bknueven thanks for demonstrating how code should be tested in documentation!

**Comment by adam-a-a** [comment]
@bknueven I assumed that you wanted to add testing to the how-to for parameter sweep, so I rescinded my approval to let you do that first. If you'd rather get this merged as is though, let me know and I'll approve again.

**Review by TimBartholomew — APPROVED** [review]
LGTM

**Review by adam-a-a — APPROVED** [review]

**Pull request merged**